scripts/build.js

```diff
@@ -20,59 +20,93 @@ function escapeForRegExp(str) {
   return str.replace(/[|\\*?+$^.{}()\[\]]/g, "\\$&");
 }
 
-const extensionPath = path.resolve(
-  __dirname,
-  "../continue/extensions/vscode/src",
-);
-const extensionFilter = new RegExp(
-  "^" + escapeForRegExp(extensionPath + path.sep) + ".*\\.ts$",
-);
+const makeTransformPlugin = ({ name, basePath, substitutions }) => {
+  const absPath = path.resolve(__dirname, "..", basePath);
+  const filter = new RegExp(
+    "^" + escapeForRegExp(absPath + path.sep) + ".*\\.ts$",
+  );
 
-const namespaceSubstitutions = [
-  {
-    pattern: /^continue[.]/,
-    replacement: "granite.",
-  },
-  {
-    pattern: /^Continue[.]continue$/,
-    replacement: "redhat.granite-code",
-  },
-  {
-    pattern: /continueGUI/,
-    replacement: "graniteGUI",
-  },
-];
+  const transform = (fileContents) => {
+    // For speed, we do this just as a string replacement, rather than
+    // using ts-morph or similar. We only care about simple strings
+    // like "continue.acceptDiff", so we can simplify matching
+    // strings in the source code.
+    return fileContents.replace(/"([^"\\\r\n]+)"/g, (m, strContents) => {
+      for (const { pattern, replacement } of substitutions) {
+        strContents = strContents.replace(pattern, replacement);
+      }
+      return '"' + strContents + '"';
+    });
+  };
 
-function namespaceTransform(fileContents) {
-  // For speed, we do this just as a string replacement, rather than
-  // using ts-morph or similar. We only care about simple strings
-  // like "continue.acceptDiff", so we can simplify matching
-  // strings in the source code.
-  return fileContents.replace(/"([^"\\\r\n]+)"/g, (m, strContents) => {
-    for (const { pattern, replacement } of namespaceSubstitutions) {
-      strContents = strContents.replace(pattern, replacement);
-    }
-    return '"' + strContents + '"';
-  });
-}
+  return {
+    name: name,
+    setup(build) {
+      build.onLoad(
+        {
+          filter: filter,
+        },
+        async (args) => {
+          const text = await fs.promises.readFile(args.path, "utf8");
```

Comment:
your update-package-json script does ```update-package-json.js:const { readFile, writeFile } = require("fs/promises");
``` instead of `fs.promises.readFile` so might be better to do that for consistency

Reply:
Yeah, this is probably from some some example code somewhere. I've left it for now, since I don't think it *really* matters.